src/main/java/nextstep/subway/member/application/FavoriteService.java

```diff
@@ -0,0 +1,85 @@
+package nextstep.subway.member.application;
+
+import nextstep.subway.member.domain.Favorite;
+import nextstep.subway.member.domain.Member;
+import nextstep.subway.member.domain.MemberRepository;
+import nextstep.subway.member.dto.FavoriteRequest;
+import nextstep.subway.member.dto.FavoriteResponse;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.domain.StationRepository;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Service
+@Transactional
+public class FavoriteService {
+
+    private final MemberRepository memberRepository;
+    private final StationRepository stationRepository;
+
+    public FavoriteService(MemberRepository memberRepository, StationRepository stationRepository) {
+        this.memberRepository = memberRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    /**
+     * 즐겨찾기를 추가합니다.
+     * @param id
+     * @param favoriteRequest
+     * @return 추가 된 즐겨찾기 ID
+     */
+    public Long addFavorite(Long id, FavoriteRequest favoriteRequest) {
+        Member member = this.memberRepository.findById(id).orElseThrow(IllegalArgumentException::new);
+        Favorite favorite = this.toFavorite(member, favoriteRequest);
+        member.addFavorite(favorite);
+        this.memberRepository.flush();
```

Comment:
flush 보다는 save를 호출하면 어떨까요?

`flush` 보다 `save`가 메소드의 목적에 더 맞는 키워드지 않을까 싶어요

```suggestion
        this.memberRepository.save(member);
```

Reply:
아직 JPA에 대한 이해가 부족하여
FavoriteServiceTest 부분에서 지연로딩으로 인해서 오류가 발생하여, 해당 테스트 쪽에 @Transactional을 사용하였는데,
이 과정에서 해당 부분을 save하니 값이 getFavorites()의 사이즈 체크 부분이 사이즈가 2가 나와서 flush로 변경하였었습니다...
해당 부분 및 테스트 부분을 수정하였습니다.